### ISSUE TITLE
database update issue

### ISSUE THREAD

**Issue by migolovanov** [go-gitea/gitea#4494]
When i tried to apply v1.5.0-rc2, got issue while starting gitea with following error:

`2018/07/21 18:23:28 [I] Log Mode: File(Info)
2018/07/21 18:23:28 [I] XORM Log Mode: File(Info)
2018/07/21 18:23:28 [I] Cache Service Enabled
2018/07/21 18:23:28 [I] Session Service Enabled
2018/07/21 18:23:28 [I] Migration: remove commits and settings unit types
2018/07/21 18:23:28 [...itea/routers/init.go:60 GlobalInit()] [E] Failed to initialize ORM engine: migrate: do migrate: Error 1054: Unknown column 'index' in 'field list'`

**Comment by lafriks** [comment]
From what version are you upgrading?

**Comment by migolovanov** [comment]
I migrated from gogs to 1.0.0, then updated to 1.4.3 and then to 1.5.0-rc1.
Database is mysql and since i enabled indexing i see "index" column in follwing tables: commit_status, issue, pull_request

**Comment by lafriks** [comment]
Do you see in xorm.log at what SQL it fails?

**Comment by migolovanov** [comment]
Yes there are following requests:
```
2018/07/21 19:02:49 [I] PING DATABASE mysql
2018/07/21 19:02:49 [I] [SQL] SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=? []interface {}{"gogs", "version"}
2018/07/21 19:02:49 [I] [SQL] SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = ? AND `TABLE_NAME` = ? AND `COLUMN_NAME` = ? [gogs version id]
2018/07/21 19:02:49 [I] [SQL] SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = ? AND `TABLE_NAME` = ? AND `COLUMN_NAME` = ? [gogs version version]
2018/07/21 19:02:49 [I] [SQL] SELECT `id`, `version` FROM `version` WHERE `id`=? LIMIT 1 []interface {}{1}
2018/07/21 19:02:49 [I] [SQL] SELECT `id`, `unit_types` FROM `team` LIMIT 100
2018/07/21 19:02:49 [I] [SQL] DELETE FROM `repo_unit` WHERE `type` IN (?,?) []interface {}{4, 7}
2018/07/21 19:02:49 [I] [SQL] UPDATE `repo_unit` SET `type` = `type` - ?, `index` = `index` - ? WHERE (`type` > ?) []interface {}{1, 1, 7}
```

**Comment by lafriks** [comment]
What is result for such query in database?
```
SELECT `id`, `version` FROM `version` WHERE `id`=1 LIMIT 1;
```

**Comment by migolovanov** [comment]
```
MariaDB [gogs]> SELECT `id`, `version` FROM `version` WHERE `id`=1 LIMIT 1;
+----+---------+
| id | version |
+----+---------+
|  1 |      38 |
+----+---------+
1 row in set (0.00 sec)
```

**Comment by lafriks** [comment]
Did upgrade to 1.4.3 was successful? Version number should be much higher for it...

**Comment by migolovanov** [comment]
Yes, it worked fine even on 1.5.0-rc1. But i guess the issue is related to code search which i enabled in config.

**Comment by migolovanov** [comment]
I believe i fixed this issue by restoring original DB but updating version number to some higher value (have no idea when it went wrong). By the way, which version number is relevant to 1.5.0-rc2 in db?

**Comment by lafriks** [comment]
It should be 69 I think

**Comment by ptman** [comment]
is this the only thing blocking 1.5.0 release? did anyone reproduce?

**Comment by techknowlogick** [comment]
@ptman there have been a couple more PRs merged into the 1.5 branch which means there will be likely another RC made before 1.5 gets released.

**Comment by techknowlogick** [comment]
Closing this issue as we have been unable to reproduce (we have since implemented migration tests into each CI run).
